Dependencies/xr/Source/ARKit/XR.mm

```diff
@@ -539,9 +624,14 @@ bool TryInitialize() {
             , getXRView{ [windowProvider{ std::move(windowProvider) }] { return (__bridge MTKView*)windowProvider(); } }
             , metalDevice{ (__bridge id<MTLDevice>)graphicsContext } {
 
-            // Create the ARSession enable plane detection, and disable lighting estimation.
+            // Create the ARSession enable plane detection, include scene reconstruction mesh if supported, and disable lighting estimation.
             session = [ARSession new];
             auto configuration = [ARWorldTrackingConfiguration new];
+            if (@available(iOS 13.4, *)) {
+                if([ARWorldTrackingConfiguration supportsSceneReconstruction: ARSceneReconstructionMesh]) {
+                    configuration.sceneReconstruction = ARSceneReconstructionMesh;
+                }
+            }
```

Comment:
Does this have a performance/memory impact?  Can we add it after the session is already started, and only enable it when the user requests mesh detection?

Reply:
Based on the documentation that I've been referencing, the scene reconstruction was always added to the config before the session starts. I'm not entirely sure how I would enable this after the session has started.

Reply:
Interesting, it would be good to understand the perf impact if this is basically always enabled. @chrisfromwork could it make sense to also allow enabling mesh detection either at the `System` level or as a parameter when starting an XR session? It could still be available as an API at the session level for platforms that support dynamically enabling/disabling it, but it would provide a way to enable/disable at a higher level as well. For HoloLens for example this would mean you could enable mesh detection before the session actually starts, and you could disable it after the session starts, whereas on iOS if you try to enable/disable after the session starts it would just return false (it can only be enabled/disabled before the session starts). Would something like that make sense?

Reply:
So, to measure the execution time I added NSDate *methodStart = [NSDate date]; at  line 1452, and NSDate *methodFinish = [NSDate date]; at line 1553 right before the call to GetNextFrame. Also added NSTimeInterval executionTime = [methodFinish timeIntervalSinceDate:methodStart]; to get the total execution time. It doesn’t seem like theres a huge difference between the timing. Let me know if there was a different way you wanted me to measure this @ryantrem 
![tempImage3Tn0kQ](https://user-images.githubusercontent.com/86322959/125369152-bd1f6080-e349-11eb-96c3-c65527d8b30b.gif)
![tempImagecrtLXJ](https://user-images.githubusercontent.com/86322959/125369163-c0b2e780-e349-11eb-9db4-bb4a8f905614.gif)